core/arch/arm/kernel/mutex.c

```diff
@@ -38,6 +38,9 @@ void mutex_init(struct mutex *m)
 
 static void __mutex_lock(struct mutex *m, const char *fname, int lineno)
 {
+	assert_have_no_spinlock();
```

Comment:
same issue: called before IRQs are masked: may switch of execution core while being executed.